hazelcast/src/test/java/com/hazelcast/test/compatibility/SamplingSerializationService.java

```diff
@@ -0,0 +1,255 @@
+        return (T) delegate.readObject(in);
+    }
+
+    @Override
+    public <T> T readObject(ObjectDataInput in, Class aClass) {
+        return (T) delegate.readObject(in, aClass);
+    }
+
+    @Override
+    public <T> T toObject(Object data) {
+        return (T) delegate.toObject(data);
+    }
+
+    @Override
+    public <T> T toObject(Object data, Class klazz) {
+        return (T) delegate.toObject(data, klazz);
+    }
+
+    @Override
+    public byte[] toBytes(Object obj) {
+        byte[] bytes = delegate.toBytes(obj);
+        sampleObject(obj, bytes);
+        return bytes;
+    }
+
+    @Override
+    public byte[] toBytes(int padding, Object obj) {
+        byte[] bytes = delegate.toBytes(padding, obj);
+        sampleObject(obj, bytes);
+        return bytes;
+    }
+
+    @Override
+    public byte[] toBytes(Object obj, PartitioningStrategy strategy) {
+        byte[] bytes = delegate.toBytes(obj, strategy);
+        sampleObject(obj, bytes);
+        return bytes;
+    }
+
+    @Override
+    public byte[] toBytes(int padding, Object obj, PartitioningStrategy strategy) {
+        byte[] bytes = delegate.toBytes(padding, obj, strategy);
+        sampleObject(obj, bytes);
+        return bytes;
+    }
+
+    @Override
+    public ClassLoader getClassLoader() {
+        return delegate.getClassLoader();
+    }
+
+    @Override
+    public BufferObjectDataInput createObjectDataInput(byte[] data) {
+        return delegate.createObjectDataInput(data);
+    }
+
+    @Override
+    public BufferObjectDataInput createObjectDataInput(Data data) {
+        return delegate.createObjectDataInput(data);
+    }
+
+    @Override
+    public BufferObjectDataOutput createObjectDataOutput(int size) {
+        return delegate.createObjectDataOutput(size);
+    }
+
+    @Override
+    public ManagedContext getManagedContext() {
+        return delegate.getManagedContext();
+    }
+
+    @Override
+    public PortableReader createPortableReader(Data data)
+            throws IOException {
+        return delegate.createPortableReader(data);
+    }
+
+    @Override
+    public PortableContext getPortableContext() {
+        return delegate.getPortableContext();
+    }
+
+    @Override
+    public void disposeData(Data data) {
+        delegate.disposeData(data);
+    }
+
+    @Override
+    public BufferObjectDataOutput createObjectDataOutput() {
+        return delegate.createObjectDataOutput();
+    }
+
+    @Override
+    public ByteOrder getByteOrder() {
+        return delegate.getByteOrder();
+    }
+
+    @Override
+    public byte getVersion() {
+        return delegate.getVersion();
+    }
+
+    @Override
+    public void dispose() {
+        delegate.dispose();
+    }
+
+    // record the given object, then return it
+    private static <T> T sampleObject(T obj, byte[] serializedObject) {
+        if (obj == null)
+            return null;
+
+        if (serializedObject != null && shouldAddSerializedSample(obj)) {
```

Comment:
I class name is checked here if should it be serialized or not. If 2 class names comes consecutively, second one will return false here, because of following check
```
if (SAMPLED_CLASSES.contains(className)) {
            return false;
      }
```

So how there can be a second object in the SERIALIZED_SAMPLES_PER_CLASS_NAME list corresponding to same class name ? As far as I understand only `MAX_SERIALIZED_SAMPLES_PER_CLASS` is meant to check. And the check from `SAMPLED_CLASS` should be removed, right ?